src/IntercomContacts.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Intercom;
+
+use Http\Client\Exception;
+use stdClass;
+
+class IntercomContacts extends IntercomResource
+{
+    /**
+     * Creates a Contact.
+     *
+     * @see    https://developers.intercom.com/intercom-api-reference/reference#create-contact
+     * @param  array $options
+     * @return stdClass
+     * @throws Exception
+     */
+    public function create(array $options)
+    {
+        return $this->client->post("contacts", $options);
+    }
+
+    /**
+     * Updates a Contact.
+     *
+     * @see    https://developers.intercom.com/intercom-api-reference/reference#update-contact
+     * @param  array $options
+     * @return stdClass
+     * @throws Exception
+     */
+    public function update(array $options)
+    {
+        return $this->client->put("contacts", $options);
+    }
+
+    /**
+     * Lists Contacts.
+     *
+     * @see    https://developers.intercom.com/intercom-api-reference/reference#list-contacts
+     * @param  array $options
+     * @return stdClass
+     * @throws Exception
+     */
+    public function getContacts(array $options)
+    {
+        return $this->client->get('contacts', $options);
+    }
+
+    /**
+     * Gets a single Contact based on the Intercom ID.
+     *
+     * @see    https://developers.intercom.com/intercom-api-reference/reference#get-contact
+     * @param  string $id
+     * @param  array  $options
+     * @return stdClass
+     * @throws Exception
+     */
+    public function getContact($id, $options = [])
+    {
+        $path = $this->contactPath($id);
+        return $this->client->get($path, $options);
+    }
+
+    /**
+     * Permenently Deletes a single Contact based on the Intercom ID.
+     *
+     * @see    https://developers.intercom.com/intercom-api-reference/reference#delete-contact
+     * @param  string $id
+     * @param  array  $options
+     * @return stdClass
+     * @throws Exception
+     */
+    public function deleteContact(string $id, array $options = [])
+    {
+        $path = $this->contactPath($id);
+        return $this->client->delete($path, $options);
+    }
+
+    /**
+     * Returns list of Contacts that match search query.
+     *
+     * @see     https://developers.intercom.com/reference#search-for-contacts
+     * @param   array $options
+     * @return  stdClass
+     * @throws  Exception
+     */
+    public function search(array $options)
+    {
+        $path = 'contacts/search';
+        return $this->client->post($path, $options);
+    }
+
+    /**
+     * Returns next page of Contacts that match search query.
+     *
+     * @see     https://developers.intercom.com/intercom-api-reference/reference#pagination-search
+     * @param   array $options
+     * @return  stdClass
+     * @throws  Exception
+     */
+    public function nextSearch(array $query, $pages)
```

Comment:
To make this simpler to reason about, why don't we pass the whole response here instead of `$pages`? Then we can handle that complexity inside this method.

Reply:
To my knowledge, the query isn't included in the response. And the pages seems consistent with how earlier versions of the pagination work within the SDK. I could changed this to the response but unless I'm misunderstanding something, the query would still need to be passed.

Reply:
My bad, I misread the docs above 🤦‍♂️ You're totally right. Let's ignore this comment 😄